compiler/rustc_trait_selection/src/traits/auto_trait.rs

```diff
@@ -292,7 +292,7 @@ impl<'tcx> AutoTraitFinder<'tcx> {
                 new_env,
                 pred,
             ));
-            let result = select.select(&obligation);
+            let result = select.poly_select(&obligation);
```

Comment:
feels like we may end up with with placeholders in the environment/output here, so it might be safer to check for `no_bound_vars`. Doesn't matter much, this keeps the current behavior and we're going to rewrite this code anyways